### DISCUSSION TITLE
global IPv6 address and hostname discovery from outside of mesh

### DISCUSSION THREAD

**Discussion by wallybarnum** [openthread/openthread#8288] [Q&A] [answered]
I'm going to run a connectivity test with 10's of Nordic 802.15.4 dongles. They are hardcoded with mesh credentials and join my (eero) border router as MTD's.

I need to discover their global IPv6 address so I can connect to/ping them.  

I can build the code to be a SRP client that registers and publishes a service, and then I see the service info with hostname.local and IPv6 address with an mDNS browser on my Wi-Fi network.

Is there a better or more built in way to accomplish IPv6 address discovery from outside the mesh?  The SRP client feature looks like it is new in Thread 1.3(?), and an off the shelf mdns browser does not show any of the mesh devices unless I publish my own service.

Thanks,
Wally

**Comment by jwhui** [comment, marked as answer]
You already discovered the primary mechanism for discovery in Thread :)

DNS-based Service Discovery is the primary mechanism for discovering services offered by Thread devices and the IPv6 addresses associated with them. It is one of the new features in Thread 1.3.

Thread devices must explicitly publish services, otherwise they will not publish any DNS-SD information.

**Reply by wallybarnum** [reply]
Ok thanks Johnathan.  I thought there was something else before, but was wrong I guess. My previous thread work was pretty much on-mesh and had a diy gateway vs a real border router.
Thanks for the help. 
Wally
